1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,35 @@
-function setAlarm() {}
+function dispalyRemainingTime(totalSeconds) {
+  let timeRemaining = document.getElementById("timeRemaining");
+  let minutes = Math.floor(totalSeconds / 60);
+  let seconds = totalSeconds % 60;
+
+  // timeRemaining.textContent = "Time Remaining  " + minutes + ":" + seconds;
+
+  if (minutes < 10) {
+    timeRemaining.textContent =
+      "Time Remaining  " + "0" + minutes + ":" + seconds;
+  } else if (seconds < 10) {
+    timeRemaining.textContent =
+      "Time Remaining  " + minutes + ":" + "0" + seconds;
+  } else {
+    timeRemaining.textContent = "Time Remaining  " + minutes + ":" + seconds;
+  }
+}
+
+function setAlarm() {
+  let inputTime = document.getElementById("alarmSet");
+  let totalSeconds = inputTime.value;
+  dispalyRemainingTime(totalSeconds);
+
+  let Interval = setInterval(() => {
+    totalSeconds -= 1;
+    dispalyRemainingTime(totalSeconds);
+    if (totalSeconds === 0) {
+      playAlarm();
+      clearInterval(Interval);
+    }
+  }, 1000);
+}
```

Comment:
nice work! 💪
The basic functionality is there, which is great!
Two things to improve:
- The timer displays only one digit in the seconds place when it gets below ten (try having a look at the .padStart and .padEnd string methods to make this easier
- You can create more than one timer if you hit the button multiple times - try to figure out a way to clear your interval somewhere else so that this doesn't happen.